### PR TITLE
Build CI

### PR DESCRIPTION
**Pull request by TheSeems**
Add simple build CI for Linux.

### CODE REVIEW

**Comment by MomoDeve** [comment]
Let'a add building with gcc to CI before merging this into master

**Comment by TheSeems** [comment]
g++ CI build support added, i hope...

**Comment by MomoDeve** [comment]
Yeah, its working. Lets also add MSVC to support all three popular compilers

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+language: cpp
+dist: bionic
+git:
+  submodules: false
+  
+matrix:
+  include:
+  - os: linux
```
Comment (WhiteBlackGoose) [inline .travis.yml:8]: Why only linux?

File: src/Utilities/ECS/Component.h [hunk 2]
```diff
@@ -63,9 +63,9 @@ namespace MxEngine
     public:
         ComponentManager() = default;
         ComponentManager(const ComponentManager&) = delete;
-        ComponentManager(ComponentManager&&) noexcept = default;
+        ComponentManager(ComponentManager&&) = default;
         ComponentManager& operator=(const ComponentManager&) = delete;
-        ComponentManager& operator=(ComponentManager&&) noexcept = default;
+        ComponentManager& operator=(ComponentManager&&) = default;
```
Comment (WhiteBlackGoose) [inline src/Utilities/ECS/Component.h:68]: 🦆🦆🦆🦆🦆

File: src/Utilities/UUID/UUID.h [hunk 3]
```diff
@@ -69,8 +68,7 @@ namespace MxEngine
     struct UUIDGeneratorImpl
     {
         using type = uuids::basic_uuid_random_generator<Random::Generator>;
-        constexpr static size_t StorageSize = sizeof(std::uniform_int_distribution<uint32_t>) + sizeof(std::shared_ptr<Random::Generator>);
-        std::aligned_storage_t<StorageSize> generator;
+        std::aligned_storage_t<24> generator;
```
Comment (WhiteBlackGoose) [inline src/Utilities/UUID/UUID.h]: 🦆🦆🦆🦆🦆

File: .travis.yml (hunk 1 above)
Comment (TheSeems) [reply .travis.yml:8]: Please, read the comment above. We're going to support build not only for linux. Converting pull to draft at the moment.

**Pull request merged**